lib/writeUris.js

```diff
@@ -48,72 +48,110 @@ function writeUris(gltf, gltfPath, resourcesDirectory, options) {
+function removeEmpty(json) {
+    Object.keys(json).forEach(function(key) {
+        if (!defined(json[key]) || (Array.isArray(json[key]) && json[key].length === 0)) {
+            delete json[key]; // Delete values that are undefined or []
+        } else if (typeof json[key] === 'object') {
+            removeEmpty(json[key]);
+        }
+    });
+}
+
+function cleanup(gltf) {
+    removeEmpty(gltf);
+}
+
+function writeSeparateBuffer(gltf, gltfPath, name) {
+    var buffer = gltf.buffers[0];
     var source = buffer.extras._obj2gltf.source;
     var bufferUri = name + '.bin';
     buffer.uri = bufferUri;
-    var bufferPath = path.join(resourcesDirectory, bufferUri);
+    var bufferPath = path.join(path.dirname(gltfPath), bufferUri);
     return fsExtra.outputFile(bufferPath, source);
 }
 
-function writeSeparateTextures(gltf, resourcesDirectory) {
+function writeSeparateTextures(gltf, gltfPath) {
     var images = gltf.images;
-    return Promise.map(Object.keys(images), function(id) {
-        var image = images[id];
+    return Promise.map(images, function(image) {
         var extras = image.extras._obj2gltf;
         var imageUri = image.name + extras.extension;
         image.uri = imageUri;
-        var imagePath = path.join(resourcesDirectory, imageUri);
+        var imagePath = path.join(path.dirname(gltfPath), imageUri);
         return fsExtra.outputFile(imagePath, extras.source);
     }, {concurrency : 10});
 }
 
 function writeEmbeddedBuffer(gltf) {
-    var buffer = gltf.buffers[Object.keys(gltf.buffers)[0]];
+    var buffer = gltf.buffers[0];
     var source = buffer.extras._obj2gltf.source;
     buffer.uri = 'data:application/octet-stream;base64,' + source.toString('base64');
 }
 
 function writeEmbeddedTextures(gltf) {
+    var buffer = gltf.buffers[0];
+    var bufferExtras = buffer.extras._obj2gltf;
+    var bufferSource = bufferExtras.source;
     var images = gltf.images;
-    for (var id in images) {
-        if (images.hasOwnProperty(id)) {
-            var image = images[id];
-            var extras = image.extras._obj2gltf;
-            image.uri = 'data:' + mime.lookup(extras.extension) + ';base64,' + extras.source.toString('base64');
-        }
+    var imagesLength = images.length;
+    var sources = [bufferSource];
+    var byteOffset = bufferSource.length;
+
+    for (var i = 0; i < imagesLength; ++i) {
+        var image = images[i];
+        var extras = image.extras._obj2gltf;
+        var imageSource = extras.source;
+        var imageByteLength = imageSource.length;
+
+        image.mimeType = mime.lookup(extras.extension);
+        image.bufferView = gltf.bufferViews.length;
+        gltf.bufferViews.push({
+            buffer : 0,
+            byteOffset : byteOffset,
+            byteLength : imageByteLength
+        });
+        byteOffset += imageByteLength;
+        sources.push(imageSource);
     }
+
+    var source = getBufferPadded(Buffer.concat(sources));
```

Comment:
I don't think they need to be. I forget the exact rules but I think byte alignment is only required for buffer views holding accessor data.